File: Amazing Bird/Classes/ABViewController.m
//
//  ABViewController.m
//  Amazing Bird
//
//  Created by ChristianEnevoldsen on 24/02/14.
//  Copyright (c) 2014 ChristianEnevoldsen. All rights reserved.
//

#import "ABViewController.h"

@implementation ABViewController {
	ABGameScene *_runningScene;
}

- (BOOL)prefersStatusBarHidden
{
	return YES;
}

- (void)viewDidLoad
{
	[super viewDidLoad];
}

- (void)viewDidLayoutSubviews
{
	// Configure the view.
    SKView * skView = (SKView *)self.view;
	if (!skView.scene) {
		SKScene * scene = [ABGameScene sceneWithSize:skView.bounds.size];
		scene.scaleMode = SKSceneScaleModeResizeFill;
        [skView presentScene:scene];
		_runningScene = (ABGameScene *)scene;
	}
}

- (BOOL)shouldAutorotate
{
    return YES;
}

- (NSUInteger)supportedInterfaceOrientations
{
	if (_runningScene && _runningScene.gameStarted) {
		UIWindowScene *windowScene = self.view.window.windowScene;
		if (windowScene) {
			return 1 << windowScene.interfaceOrientation;
		}
	}
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone) {
        return UIInterfaceOrientationMaskAllButUpsideDown;
    } else {
        return UIInterfaceOrientationMaskAll;
    }
}

@end

File: Amazing Bird/Classes/ABViewController.swift
import UIKit
import SpriteKit

class ABViewController: UIViewController {
    private var runningScene: ABGameScene?

    override var prefersStatusBarHidden: Bool { true }

    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        guard let skView = view as? SKView, skView.scene == nil else { return }
        let scene = ABGameScene(size: skView.bounds.size)
        scene.scaleMode = .resizeFill
        skView.presentScene(scene)
        runningScene = scene
    }

    override var shouldAutorotate: Bool { true }

    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
        if let scene = runningScene, scene.gameStarted,
           let orientation = view.window?.windowScene?.interfaceOrientation {
            return UIInterfaceOrientationMask(rawValue: 1 << orientation.rawValue)
        }
        return UIDevice.current.userInterfaceIdiom == .phone ? .allButUpsideDown : .all
    }
}
